### ISSUE TITLE
o:commandScript causes f:viewAction actions to be re-run on render="@all" submissions

### ISSUE THREAD

**Issue by wsaxton** [omnifaces/omnifaces#254]
I'm not sure if this is a bug or working as intended, but I have a page with a f:viewAction that processes some parameters.  When I do regular submissions on the page, everything works fine (ie the viewAction action is only executed when the page is initially loaded).

When I add a o:commandScript/h:outputScript to load some data after the page is loaded, then any submissions I perform on the page re-execute the f:viewAction action every time:

testClassOmni.xhtml:

```
<?xml version="1.0" encoding="UTF-8"?>
<html xmlns="http://www.w3.org/1999/xhtml"
      xmlns:f="http://xmlns.jcp.org/jsf/core" 
      xmlns:h="http://xmlns.jcp.org/jsf/html" 
      xmlns:o="http://omnifaces.org/ui">
    <h:head>
        <f:metadata>
            <f:viewAction action="#{testClassOmni.processParams}" />
        </f:metadata>
    </h:head>
    <h:body>
        <h:form>
            
            <!-- These 2 entries cause processParams to be run when the button is pressed.
                  Remove these, and the button process does not run processParams -->
            <o:commandScript name="runCommandScript" action="#{testClassOmni.runCommandScript}"/>
            <h:outputScript target="body">runCommandScript()</h:outputScript>
            
            <h:commandButton action="#{testClassOmni.pushButton}">
                <f:ajax execute="@this" render="@all"/>
            </h:commandButton>
            
        </h:form>
    </h:body>
</html>
```

TestClassOmni.java:

```
package com.example.ztestwebvanilla;

import java.io.Serializable;
import javax.inject.Named;
import javax.faces.view.ViewScoped;

@Named
@ViewScoped
public class TestClassOmni implements Serializable {

    public void processParams() {        
        System.out.println("processParams omni");        
    }
    
    public void runCommandScript() {
        System.out.println("runCommandScript omni");                
    }
    
    public void pushButton() {
        System.out.println("pushButton omni");
    }
}
```

Output when the page is loaded and the button is pressed a few times:

> Info:   processParams omni
> Info:   runCommandScript omni
> Info:   pushButton omni
> Info:   processParams omni
> Info:   pushButton omni
> Info:   processParams omni

Output when the the o:commandScript is removed:

> Info:   processParams omni
> Info:   pushButton omni
> Info:   pushButton omni

**Comment by BalusC** [comment]
Can't reproduce this. This output is however unexpected as I'd expect the runCommandScript to be re-executed on `@all` instead of processParams. This is the (technically expected) output I get:

> processParams omni
> runCommandScript omni
> pushButton omni
> runCommandScript omni
> pushButton omni
> runCommandScript omni

On both Mojarra 2.2.13 and MyFaces 2.2.10.

**Comment by wsaxton** [comment]
Hmm, I'm running this on Glassfish 4.1 using Mojarra 2.2.7.  Will look into figuring out how to upgrade GF to use 2.2.13 and re-test.

**Comment by wsaxton** [comment]
I upgraded to using Payara 4.1.1.161.1 which contains Mojarra 2.2.12 and I still get the output I posted.

I've attached the Netbeans project (which contains the .war in target/ZTestWebVanilla-1.0-SNAPSHOT.war in case you want to reproduce.

[ZTestWebVanilla.zip](https://github.com/omnifaces/omnifaces/files/276102/ZTestWebVanilla.zip)

**Comment by BalusC** [comment]
Still can't reproduce it based on given source files.

As I'm using Eclipse, I couldn't flawlessly import your Netbeans-generated zip. I had to extract it and include the invidivual files. I also had to change the `pom.xml` to remove `maven-dependency-plugin`  (Eclipse showed an error that it doesn't recognize its execution). That plugin looked weird to me, it seems to include a Java EE endorsed API JAR file somewhere. If it indeed actually ended up in runtime classpath, it might explain your problem.
